packages/frontend/src/presentation/screens/CardSearch/CardSearchFilter.tsx

```diff
@@ -4,21 +4,58 @@ import { useIsMounted } from '../../hooks/IsMounted';
 import { NavigationScreenProp, NavigationState } from 'react-navigation';
 import { ROUTES } from '../../Navigator';
 import { CardSearchFilterLayout } from './CardSearchFilterLayout';
+import { Card } from '../../../domain/entities/Card';
 
 interface Props {
   navigation: NavigationScreenProp<NavigationState>;
 }
 
 export const CardSearchFilter: React.FC<Props> = (props: Props) => {
   const [cardName, setCardName] = useState('');
+  const [supertype, setSupertype] = useState('');
+  const [subtype, setSubtype] = useState('');
+  const [colors, setColors] = useState(['']);
+  const [colorIdentities, setColorIdentities] = useState(['']);
   const isMounted = useIsMounted();
 
+  function byColor(card: Card): boolean | Card {
+    if (!colors.filter(color => !!color).length) {
+      return card;
+    }
+    return card.colors.some(color => colors.includes(color));
+  }
+
+  function byColorIdentity(card: Card): boolean | Card {
+    if (!colorIdentities.filter(color => !!color).length) {
+      return card;
+    }
+    return card.colorIdentities.some(color => colorIdentities.includes(color));
+  }
+
+  function bySubtype(card: Card): boolean | Card {
+    if (!subtype) {
+      return card;
+    }
+    return card.subTypes ? card.subTypes.includes(subtype) : card;
+  }
+
+  function bySupertype(card: Card): boolean | Card {
+    if (!supertype) {
+      return card;
+    }
+    return card.superTypes ? card.superTypes.includes(supertype) : card;
+  }
+
   async function handleSubmit(): Promise<void> {
     const cards = await querySampleCardList();
 
     if (isMounted()) {
       props.navigation.navigate(ROUTES.CARD_SEARCH_RESULTS, {
-        cardsFiltered: cards,
+        cardsFiltered: cards
+          .filter(bySubtype)
```

Comment:
Acredito que essa forma de tratar os filtros nao vai escalar muito bem, mas ainda nao to conseguindo pensar em uma maneira mais eficaz de fazer os filtros.

Reply:
Filtro de carta vai ter que ser um modulo a parte.
Algo com a seguinte assinatura (Exemplo ludico):
```ts
CardsManager.searchBy(filter: CardFilter): Card[]
```

Ai a logica e complexidade de se procurar uma carta sai da camada de apresentacao.